spring-scheduling/src/test/java/com/baeldung/asyncwithretry/EventServiceIntegrationTest.java

```diff
@@ -0,0 +1,90 @@
+package com.baeldung.asyncwithretry;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+import org.springframework.web.client.HttpServerErrorException;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {AsyncConfig.class}, loader = AnnotationConfigContextLoader.class)
+class EventServiceIntegrationTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventServiceIntegrationTest.class);
+
+    @Autowired
+    private EventService asyncEventService;
+
+    @MockBean
+    private DownstreamService downstreamService;
+
+    @Test
+    void givenAsyncMethodHasNoException_whenAsyncMethodIscalled_thenReturnSuccess_WithoutAnyRetry() throws InterruptedException, ExecutionException {
```

Comment:
We don't need to throw multiple checked exceptions here - just use `throws Exception` from any test

Reply:
sure, i've updaed with Exception where required multiples exception type.